clientv3/cluster.go

```diff
@@ -36,8 +36,12 @@ type Cluster interface {
 	// MemberList lists the current cluster membership.
 	MemberList(ctx context.Context) (*MemberListResponse, error)
 
-	// MemberAdd adds a new member into the cluster.
-	MemberAdd(ctx context.Context, peerAddrs []string) (*MemberAddResponse, error)
+	// MemberAddAsAutoPromoting adds a new member as a learner that is
+	// automatically promoted to a node upon catching up with the leader into the cluster.
+	MemberAddAsAutoPromotingNode(ctx context.Context, peerAddrs []string) (*MemberAddResponse, error)
+
+	// MemberAddAsNode adds a new member as a node into the cluster.
+	MemberAddAsNode(ctx context.Context, peerAddrs []string) (*MemberAddResponse, error)
```

Comment:
I think we should keep MemberAdd() unchanged for better backward compatibility.

Reply:
Good point, done in 29fe958